org.andrew.ytdownloader.yml

```diff
@@ -0,0 +1,51 @@
+app-id: org.andrew.ytdownloader
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node18
+command: run.sh
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --filesystem=home
+  - --talk-name=org.freedesktop.Notifications
+  - --share=network
+build-options:
+  append-path: /usr/lib/sdk/node18/bin
+  env:
+    NPM_CONFIG_LOGLEVEL: info
+modules:
+  - name: ytdownloader
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/ytdownloader/flatpak-node/cache
+        npm_config_cache: /run/build/ytdownloader/flatpak-node/npm-cache
+        npm_config_offline: 'true'
+    build-commands:
+      # Install npm dependencies
+      - npm install --offline
+      # Build the app; in this example the `dist` script
+      # in package.json runs electron-builder
+      - |
+        . ../flatpak-node/electron-builder-arch-args.sh
+        npm run dist -- $ELECTRON_BUILDER_ARCH_ARGS  --linux --dir
+      # Bundle app and dependencies
+      - cp -a dist/linux*unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../run.sh
+    subdir: main
+    sources:
+      - type: repo
+        url: https://github.com/aandrew-me/ytdownloader-flatpak.git
```

Comment:
you need an actual git commit.

also why aren't you using the original repo https://github.com/aandrew-me/ytDownloader ?

Reply:
What do you mean by 'actual git commit'?

Because for some reason the app doesn't build with ffmpeg-static but builds with ffmpeg-static-electron when using flatpak-builder.

Reply:
this is not git commit or tag specified. you need one.

Reply:
Could you please clarify that one or point me to some documentation. Thanks

Reply:
https://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html

Reply:
I don't need to export to a git repo, I need the files to be downloaded from there.

Reply:
**After adding generated-sources.json and changing branch name from main to master, I am able to successfully build the app.**

Reply:
you still need a commit or a tag here.

Reply:
How do I add a commit?

Reply:
I linked above to the documentation as requested: https://github.com/flathub/flathub/pull/3390#discussion_r938834308

I would have linked to the exact paragraph but it doesn't allow this. Search for `commit`.

Reply:
Even after reading I can't seem to understand how to include commit/tag. There is an example provided which has git but there is not commit there. Should this commit be added to manifest file? If yes should it be under sources?